src/main/java/org/generationcp/middleware/service/impl/study/StudyEntryServiceImpl.java

```diff
@@ -173,27 +174,18 @@ private void addStudyEntryPropertyDataIfApplicable(final StockModel stock, final
 	}
 
 	@Override
-	public void updateStudyEntryProperty(final int studyId, final StudyEntryPropertyData studyEntryPropertyData) {
-		final StockProperty stockProperty = this.daoFactory.getStockPropertyDao().getById(studyEntryPropertyData.getStudyEntryPropertyId());
-		if (stockProperty != null) {
-			stockProperty.setValue(studyEntryPropertyData.getValue());
-			this.daoFactory.getStockPropertyDao().saveOrUpdate(stockProperty);
-		}
+	public void updateStudyEntriesProperty(final StudyEntryPropertyDataUpdateRequestDto studyEntryPropertyDataUpdateRequestDto) {
+		this.daoFactory.getStockPropertyDao().updateByStockIdsAndTypeId(studyEntryPropertyDataUpdateRequestDto.getEntryIds(),
+			studyEntryPropertyDataUpdateRequestDto.getVariableId(), studyEntryPropertyDataUpdateRequestDto.getValue());
 	}
 
 	@Override
-	public Boolean hasUnassignedEntries(final int studyId) {
-		return this.daoFactory.getStockDao().hasUnassignedEntries(studyId);
+	public List<Integer> hasPlotEntries(final List<Integer> entryIds) {
```

Comment:
if method has name hasXXXX, then usually it returns boolean type. You can rename as `getEntriesWithUsedInPlots`